avalon/maya/workio.py

```diff
@@ -30,10 +30,27 @@ def current_file():
     return current_filepath
 
 
-def work_root():
-
-    # Base the root on the current Maya workspace.
-    return os.path.join(
-        cmds.workspace(query=True, rootDirectory=True),
-        cmds.workspace(fileRuleEntry="scene")
-    )
+def work_root(session):
+    work_dir = session["AVALON_WORKDIR"]
+    scene_dir = None
+
+    # Query scene file rule from workspace.mel if it exists in WORKDIR
+    workspace_mel = os.path.join(work_dir, "workspace.mel")
+    if os.path.exists(workspace_mel):
+        scene_rule = 'workspace -fr "scene" '
+        # We need to use builtins as `open` is overridden by the workio API
+        open_file = __builtins__["open"]
+        with open_file(workspace_mel, "r") as f:
+            for line in f:
+                if line.strip().startswith(scene_rule):
+                    remainder = line[len(scene_rule):]      # == "rule";
+                    scene_dir = remainder.split('"')[1]     # == rule
```

Comment:
> But somehow when scrolling through the asset list and it doing that many times after each other often crashed Maya.

Can we add this as note comment in code ? Was afraid that I might forgot why doing this way instead of using Maya command. :baby:

Reply:
Added a comment with: https://github.com/getavalon/core/pull/442/commits/2f4aa80808010be8f8dfc10bdac54c7a05ad0a85